src/main/java/net/dv8tion/jda/internal/utils/config/ThreadingConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015-2019 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.utils.config;
+
+import net.dv8tion.jda.internal.utils.concurrent.CountingThreadFactory;
+
+import java.util.concurrent.*;
+import java.util.function.Supplier;
+
+public class ThreadingConfig
+{
+    private ScheduledExecutorService rateLimitPool;
+    private ScheduledExecutorService gatewayPool;
+    private ExecutorService callbackPool = ForkJoinPool.commonPool();
+
+    private boolean shutdownRateLimitPool = true;
+    private boolean shutdownGatewayPool = true;
+    private boolean shutdownCallbackPool = false;
```

Comment:
Use a parameterless constructor to set these. We may need to do more than this in the future, so lets do what the other configs are doing and use a constructor.

Reply:
Yeah why not